contracts/api3-server-v1/proxies/Api3CrossRateReaderProxyV1.sol

```diff
@@ -0,0 +1,193 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.27;
+
+import "../../vendor/@openzeppelin/contracts@5.0.2/proxy/utils/UUPSUpgradeable.sol";
+import "../../vendor/@openzeppelin/contracts-upgradeable@5.0.2/access/OwnableUpgradeable.sol";
+import "../../vendor/@chainlink/contracts@1.2.0/src/v0.8/shared/interfaces/AggregatorV2V3Interface.sol";
+import "./interfaces/IApi3CrossRateReaderProxyV1.sol";
+import "../../interfaces/IApi3ReaderProxy.sol";
+
+/// @title UUPS-upgradeable IApi3CrossRateReaderProxyV1 and
+/// AggregatorV2V3Interface implementation that is designed to be deployed by
+/// Api3CrossRateReaderProxyV1Factory
+/// @notice The owner of this contract is allowed to upgrade it. In the case
+/// that it is deployed by Api3CrossRateReaderProxyV1Factory, the owner will be
+/// the owner of Api3CrossRateReaderProxyV1Factory at the time of deployment.
+/// @dev For a gas-cheap `read()` implementation, this upgradeable contract
+/// uses immutable variables (rather than initializable ones). To enable this,
+/// an Api3CrossRateReaderProxyV1 needs to be deployed for each unique
+/// combination of variables. The end user does not need to concern themselves
+/// with this, as Api3CrossRateReaderProxyV1Factory abstracts this detail away.
+/// Refer to https://github.com/api3dao/migrate-from-chainlink-to-api3 for more
+/// information about the Chainlink interface implementation.
+contract Api3CrossRateReaderProxyV1 is
```

Comment:
I'm not a fan of "cross rate" in the name. I think `CompositeApi3ReaderProxyV1` would be better.

Reply:
https://github.com/api3dao/contracts/pull/472/commits/7c58ea532f319360ee8bee22982ec3b072a32f49